### ISSUE TITLE
binlog corrupt when build failed?

### ISSUE THREAD

**Issue by japj** [jeffkl/MSBuildProjectCreator#46]
Hello,

Up until now the creation of binlog files has been succesfull.
However, I have a failed build and ran into an issue that the binlog file seems to be corrupt ("again").
```
.TryBuild(restore: true, target: "SomeTarget", out result, out buildOutput);
```
result is false (because of something I want to verify) and the binlog file only shows "Build failed" and no actual MSBuild content in the binlog viewer.

**Comment by jeffkl** [comment]
Loggers aren't shutdown properly unless you dispose of the `ProjectCollection`.  The code below creates a valid `.binglog`.

```C#
string binLogPath = Path.Combine(TestRootPath, "test.binlog");

bool result;
BuildOutput buildOutput;

using (ProjectCollection projectCollection = new ProjectCollection())
{
    projectCollection.RegisterLogger(new BinaryLogger
    {
        Parameters = $"LogFile={binLogPath}",
    });

    ProjectCreator.Create(
            projectCollection: projectCollection)
        .TaskError("Failed")
        .Save(GetTempFileName(".proj"))
        .TryBuild(out result, out buildOutput);
}
```

**Comment by japj** [comment]
It might be that the "corrupt" binlog is due to the error:
```
Microsoft.Build.Exceptions.InvalidProjectFileException: The SDK 'Sdk.props' specified could not be found
```
Perhaps MSBuild is throwing this error early/at a place where the binlog cannot be properly written.

And while writing this reply I just realized I mixed the ImportSdk name and project around, so that solves part of my problem.

It still means there is a "path" where the .binlog file cannot be written correctly/contains incomplete data.
